skimage/graph/_mcp.pyx

```diff
@@ -299,7 +299,11 @@ cdef class MCP:
         # but this is my code and I like fortran-style! Also, it's
         # faster when working with image arrays, which are often
         # already fortran-strided.)
-        self.flat_costs = costs.astype(FLOAT_D).flatten('F')
+        try:
+            self.flat_costs = costs.astype(FLOAT_D, copy=False).ravel('F')
+        except TypeError:
+            self.flat_costs = costs.astype(FLOAT_D).flatten('F')
+            print('Using older Numpy version. Upgrading might decrease memory usage and increase speed.')
```

Comment:
It may be that it is not a TypeError, in which case none of these statements will execute.